core/components/header/schema.ts

```diff
@@ -0,0 +1,12 @@
+import { z } from 'zod';
+
+import type { CurrencyCode } from './fragment';
+
+// eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+export const currencyCodeSchema = z
+  .string()
+  .length(3)
+  .toUpperCase()
+  .refine((val) => /^[A-Z]{3}$/.test(val), {
```

Comment:
You can use a type guard instead of casting:
```suggestion
  .refine((val): val is CurrencyCode => /^[A-Z]{3}$/.test(val), {
```